azkaban-common/src/test/resources/azkaban/executor/container/v1PodTestTemplate1.yaml

```diff
@@ -47,15 +55,13 @@ spec:
         initialDelaySeconds: 5
         periodSeconds: 5
       volumeMounts:
+        - mountPath: /export/some/path
+          name: jobtype-volume-spark
```

Comment:
Values of container "mountPath" (`/export/some/path`) and init container "mountPath" (`/data/jobtypes/spark2`) above does not match those in the pod spec generated by Azkaban code however, the final pod spec is resolved with the latter values (`/export/apps/azkaban/azkaban-exec-server/current/plugins/jobtypes/spark` and `/data/jobtypes/spark` respectively). 
@aditya1105 Is that expected? Why the values in the template pod spec don't overwrite values in the pod spec generated by Azkaban?

Reply:
Nearly at every place, the higher priority is with the generated pod-spec rather than the template one. This is the case at many places like image path, image version, pull policy, mount path, etc.
If the volume name is the same, then the final path is overridden by the generated one.